WordPress/gutenberg
# [RNMobile] Add `enableCaching` param to fetch request on Android

`gutenberg-mobile` PR: https://github.com/wordpress-mobile/gutenberg-mobile/pull/3419
`WordPress-Android` PR: https://github.com/wordpress-mobile/WordPress-Android/pull/14529





## Description


On Android the network requests coming from RN are always [cached for 10 minutes](https://github.com/wordpress-mobile/WordPress-FluxC-Android/blob/30cab6f0045258b513e173c325b10dc25e19ee73/fluxc/src/main/java/org/wordpress/android/fluxc/network/BaseRequest.java#L36), in most cases this behavior is ok but for content that can be modified often like the reusable blocks, we should have a way to bypass the cache.

For this purpose, I've added a param to the fetch request function that is exposed to RN and also add a constant with the endpoint patterns that shouldn't be cached when a network request is made. For now we'll only disable caching for the blocks (reusable blocks) endpoint.

## How has this been tested?




1. Add a reusable block to a post/page (via the web version)
2. Open the post in the mobile app
3. Observe that the content matches what is displayed in both native and web versions
4. Edit the content of the reusable block via the web version and save the changes
5. Open the post again in the mobile app
6. Observe that the content matches what is displayed (including the changes) in both native and web versions

## Screenshots 

N/A

## Types of changes




New feature

## Checklist:
